### PR TITLE
Fix repetition of :out-dir when perun processes a file more than once

### PR DESCRIPTION
**Pull request by bhagany**
@arichiardi I think that this applies pretty much the same fix you merged in #200, but everywhere

### CODE REVIEW

**Comment by arichiardi** [comment]
Ok cool so I will test this out against the current working site I have and report back

**Comment by arichiardi** [comment]
So I checked this and now with the following:

```clojure
(def posts-dir "blog/posts")
(perun/render :out-dir posts-dir :renderer 'com.andrearichiardi.blog.post/render-page)
```

I get:

```shell
￼￼￼target/
├── blog
│   └── posts
│       └── public
│           ├── cljs-on-gh-pages.html
│           ├── cljs-replumb-require.html
│           ├── replumb-ast.html
│           └── semantic-perun-blog.html
└── public
    ├── blog
    ...
```

While before it was correctly under `public`, also note that all my posts are in `resources` (not `resources/public`). I feel this is expected so the patch seems to work for me.
Should I try something else just to make sure?

**Comment by arichiardi** [comment]
Uhm, but now if I move the posts to `public` it is not rendering them anymore.

File: src/io/perun.clj [hunk 1]
```diff
@@ -802,20 +808,17 @@
...
+  (let [{:keys [renderer out-dir] :as options} (merge +render-defaults+ *opts*)]
     (letfn [(render-paths [fileset]
               (let [entries (filter-meta-by-ext fileset options)]
                 (reduce
-                 (fn [result {:keys [out-dir filename path] :as entry}]
+                 (fn [result {:keys [path] :as entry}]
                    (let [content (slurp (boot/tmp-file (boot/tmp-get fileset path)))
-                         path-args (if (= out-dir (:out-dir options)) ;; if custom out-dir we cannot rely on pre-built :path
-                                     [path]
-                                     [out-dir (:out-dir options) filename])
```
Comment (arichiardi) [inline src/io/perun.clj:813]: this lines seems to me to build the path that would work for me, `public` before the specified `:out-dir`.

**Comment by arichiardi** [comment]
So ok, I finally settled with another solution, blog posts are in `resources/blog/posts` and the render function is a simple:

```clojure
(perun/render :renderer 'com.andrearichiardi.blog.post/render-page)
```

**Review by podviaznikov — APPROVED** [review]

**Comment by arichiardi** [comment]
It would be great to merge this one as well and maybe deploy a `SNAPSHOT` with both this and the `lessc` task :wink:

**Pull request merged**